OpenRA.Mods.Common/Activities/UnloadCargo.cs

```diff
@@ -56,40 +59,42 @@ IEnumerable<CPos> BlockedExitCells(Actor passenger)
 		public override Activity Tick(Actor self)
 		{
 			cargo.Unloading = false;
-			if (IsCanceled || cargo.IsEmpty(self))
+			if (IsCanceled || !actorsToUnload.Any() || !cargo.Passengers.Intersect(actorsToUnload).Any())
 				return NextActivity;
 
 			if (cloak != null && cloak.Info.UncloakOnUnload)
 				cloak.Uncloak();
 
-			var actor = cargo.Peek(self);
-			var spawn = self.CenterPosition;
-
-			var exitSubCell = ChooseExitSubCell(actor);
-			if (exitSubCell == null)
+			foreach (var actor in actorsToUnload)
```

Comment:
This *is* what the old code did.
It has just been put inside of a `foreach`.
I considered a delay but that is out of scope for this PR.